test/lib/indexSpec.js

```diff
@@ -43,21 +43,30 @@ describe('Logger', function() {
 
             ['debug', 'info', 'warn'].forEach(function (level) {
                 it(level + ' should log to the console via stdout', function() {
-                    logger[level]('test log message');
+                    let msg = 'test log message';
+                    logger[level](msg);
                     stdMocks.restore();
                     var output = stdMocks.flush();
                     expect(output.stdout.length).to.equal(1);
-                    var re = new RegExp(level + ': [0-9-]{10} [0-9:]{8} \\[not-set\\] test log message');
+                    var re = new RegExp([`\\[not-set]`,
+                            `\\[${level}]`,
+                            '\\d\{4\}-\\d\\d-\\d\\dT\\d\\d:\\d\\d:\\d\\d(\\.\\d+)?((\\[+-\\]\\d\\d:\\d\\d)|Z)\?',
+                            msg].join(' '));
```

Comment:
😱  😂

Reply:
Just a little question - the \\[+-\\]\\d... looks a bit funny to me. Surely the square brackets should be character set markers rather than literals as I think the iso format is '+' or '-'.  Either [+-] or (?:+|-) would do. Maybe I'm missing something?

Reply:
RegExp takes a string. When you pass in a string, it will unescape the literal markers rendering an invalid Regex. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp

Reply:
True, but the square brackets have double backslashes which un-escape to single backslashes that mean that the regex will treat them as literal '['s rather than character set markers. You don't have ^ and $ to force a complete string match so maybe it doesn't matter too much - but then 
a simpler expression would do.